### PR TITLE
path-escape-flag: add escape flag, replace backslashes with double ba…

### PR DESCRIPTION
**Pull request by atahabaki**
Related issue: #9838

Changes:

- added `--escape` flag to `path join` subcommand.
- replaces `\` with `\\`.
- Just made it available to Windows users only.

### CODE REVIEW

**Comment by amtoine** [comment]
@atahabaki 
should this close #9838? :yum:

**Comment by atahabaki** [comment]
> @atahabaki 
> should this close #9838? :yum: 

Yeah, it should close.

File: crates/nu-command/src/path/join.rs [hunk 1]
```diff
@@ -25,6 +32,20 @@ impl Command for SubCommand {
...
+                (Type::String, Type::String),
+                (Type::List(Box::new(Type::String)), Type::String),
+                (Type::Record(vec![]), Type::String),
+                (Type::Table(vec![]), Type::List(Box::new(Type::String))),
+            ])
+            .switch("escape", "use double backslash instead of one.", None)
+            .rest("append", SyntaxShape::String, "Path to append to the input")
+    }
+
+    #[cfg(not(windows))]
```
Comment (sholderbach) [inline crates/nu-command/src/path/join.rs:48]: Is it really necessary to make this a platform specific signature?

As devil's advocate is a `\ ` escape of spaces in a unix string also a `--escaped` path to join?

This leads me to question if this is the right place to make hard platform differences? `path join` inherently has a path separator platform specificity but paths are in principle just a array of bytes that have a meaning on the operating system based on the path separator that is OS specific.

Escaping is something that has a meaning both in literals and when passing quoted or unquoted strrings as arguments to external programs/protocols through a string API. Not sure if this really applies correctly to `path join`

**Review by sholderbach — CHANGES_REQUESTED** [review]

**Comment by fdncred** [comment]
Seems like we could have these changes without duplicating everything for windows or unix. 

But the real question here is whether it's better for Windows users to have `... | path join --escape` available as a QOL improvement versus having to do `... | path join | str replace '\\' '\\\\' -a` when needed. This is also being discussed here https://github.com/nushell/nushell/issues/9838.

My vote is to have `--escape` unless we can come up with another command that understand when it's appropriate to escape things.

**Pull request closed without merging**